### PR TITLE
feat: move aws->kubernetes relationship to fixtures

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: scrapers/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -172,87 +165,33 @@ func extractResults(ctx context.Context, config v1.Kubernetes, objs []*unstructu
 							awsRoleARN  = getContainerEnv(spec, "AWS_ROLE_ARN")
 							vpcID       = getContainerEnv(spec, "VPC_ID")
 							clusterName = getContainerEnv(spec, "CLUSTER_NAME")
-							awsRegion   = getContainerEnv(spec, "AWS_REGION")
```
Comment (moshloop) [inline scrapers/kubernetes/kubernetes.go:175]: Why do we remove region?

Comment (adityathebe) [reply scrapers/kubernetes/kubernetes.go:175]: it was used to form `EKSClusterKubernetesCluster` relationship and is no longer needed.

Comment (moshloop) [reply scrapers/kubernetes/kubernetes.go:175]: I think we will need to add region and availability zone name, zone id, subnet  tags so we can build those relationships

Comment (adityathebe) [reply scrapers/kubernetes/kubernetes.go:175]: @moshloop I couldn't find the Zone ID and subnet details anywhere with the kubernetes scraper. I checked the node labels & `aws-node` pod container envs

## Node Labels

image

Comment (adityathebe) [reply scrapers/kubernetes/kubernetes.go:175]: @moshloop 

Subnet to Node linking can be done through using `Subnet.VpcId`
and then Availability Zone ID to Node can be done through `ZoneID.RegionName` ?

i.e.

```yaml
relationship:
    # ZoneID to Kubernetes Node
    - filter: config_type == 'AWS::AvailabilityZoneID'
      expr: |
        [{"type": "Kubernetes::Node", "labels": {"topology.kubernetes.io/region": config["RegionName"]}}].toJSON()
    # Subnet to Kubernetes Node
    - filter: config_type == 'AWS::EC2:Subnet'
      expr: |
        [{"type": "Kubernetes::Node", "labels": {"vpc-id": config["VpcId"]}}].toJSON()
```

Comment (moshloop) [reply scrapers/kubernetes/kubernetes.go:175]: These relationships are too broad - 

Availability Zone should be a child of the Account, and then we can relate items with zone back to the Availability Zone for that account - Different accounts will have different AZ's

```
    - filter: config_type == 'AWS::AvailabilityZone'
      expr: |
        [{"type": "Kubernetes::Node", "labels": {"topology.kubernetes.io/zone": config["Name"], "accountId", config.tags['accountId'] }}].toJSON()
```

Then Availability Zone Id is linked to the Account Specific Zones. 

```
    - filter: config_type == 'AWS::AvailabilityZone'
      expr: |
        [{"type": "AWS::AvailabilityZoneId", " name": config["ZoneId"]}}].toJSON()
```

I think the region, zone, account, vpc, subnet tags will need be applied on all k8s and aws resources.

File: fixtures/aws.yaml [hunk 2]
```diff
@@ -38,6 +38,40 @@ spec:
       # - vpc
       # - SecurityGroup
       transform:
+        relationship:
+          # EKS Cluster to Kubernetes Cluster & Kubernetes Node
+          - filter: config_type == 'AWS::EKS::Cluster'
+            expr: |
+              [
+                {"type": "Kubernetes::Cluster", "labels": {"eks-cluster-name": tags["alpha.eksctl.io/cluster-name"]}},
```
Comment (moshloop) [inline fixtures/aws.yaml:46]: Missing account filters - you could have the same cluster name across multiple accounts

File: fixtures/aws.yaml [hunk 3]
```diff
@@ -38,6 +38,40 @@ spec:
...
+            expr: |
+              [{"type": "Kubernetes::Node", "labels": {"alpha.eksctl.io/instance-id": config["instance_id"]}}].toJSON()
+          # IAM Role to Kubernetes Node
+          - filter: config_type == 'AWS::IAM::Role'
+            expr: |
+              [{"type": "Kubernetes::Node", "labels": {"aws/iam-role": config["Arn"]}}].toJSON()
+          # Subnet to Kubernetes Node
+          - filter: config_type == 'AWS::EC2:Subnet'
+            expr: |
+              [{"type": "Kubernetes::Node", "labels": {"aws/account-id": tags['account'], "aws/vpc-id": config["VpcId"]}}].toJSON()
```
Comment (moshloop) [inline fixtures/aws.yaml:60]: This will match nodes in different subnets as well

File: fixtures/aws.yaml [hunk 4]
```diff
@@ -38,6 +38,40 @@ spec:
...
+            expr: |
+              [
+                {"type": "Kubernetes::Node", "labels": {"aws/account-id": tags['account'], "topology.kubernetes.io/region": config["RegionName"]}},
+                {"type": "AWS::AvailabilityZoneID", "name": config["ZoneId"]}
+              ].toJSON()
+          # Region to AvailabilityZone & ZoneID
+          - filter: config_type == 'AWS::Region'
+            expr: |
+              [
+                {"type": "AWS::AvailabilityZone", "labels": {"region": name}},
```
Comment (moshloop) [inline fixtures/aws.yaml:72]: An availability zone is an account specific thing
```suggestion           
```

File: scrapers/kubernetes/kubernetes.go [hunk 5]
```diff
@@ -146,22 +155,27 @@ func extractResults(ctx context.Context, config v1.Kubernetes, objs []*unstructu
 		}
 
 		var relationships v1.RelationshipResults
+
+		if obj.GetKind() == "Node" {
+			if val, ok := obj.GetLabels()["topology.kubernetes.io/region"]; ok {
+				labelsForIPBasedResources["aws/region"] = val
```
Comment (moshloop) [inline scrapers/kubernetes/kubernetes.go:161]: Why do we rename the label from `topology.kubernetes.io/region` to `aws/region`

Comment (adityathebe) [reply scrapers/kubernetes/kubernetes.go:161]: that label is still there. It's coming directly from the kubernetes object.

File: fixtures/aws.yaml [hunk 6]
```diff
@@ -38,6 +38,33 @@ spec:
       # - vpc
       # - SecurityGroup
       transform:
+        relationship:
+           # EKS Cluster to Kubernetes Cluster & Kubernetes Node
+          - filter: config_type == 'AWS::EKS::Cluster'
+            expr: |
+              [
+                {"type": "Kubernetes::Cluster","labels": {"aws/account-id": tags['account'],"eks-cluster-name": tags["alpha.eksctl.io/cluster-name"]}},
```
Comment (moshloop) [inline fixtures/aws.yaml:46]: wouldn't it be name instead of eks-cluster-name tag

File: fixtures/aws.yaml [hunk 7]
```diff
@@ -38,6 +38,33 @@ spec:
...
+              [{"type": "Kubernetes::Node", "labels": {"alpha.eksctl.io/instance-id": config["instance_id"]}}].toJSON()
+          # IAM Role to Kubernetes Node
+          - filter: config_type == 'AWS::IAM::Role'
+            expr: |
+              [{"type": "Kubernetes::Node", "labels": {"aws/iam-role": config["Arn"]}}].toJSON()
+          # AvailabilityZone to Zone ID & Kubernetes Node
+          - filter: config_type == 'AWS::AvailabilityZone'
+            expr: |
+              [
+                {"type": "Kubernetes::Node", "labels": {"aws/account-id": tags['account'], "topology.kubernetes.io/region": config["RegionName"]}},
```
Comment (moshloop) [inline fixtures/aws.yaml:61]:
```suggestion
                {"type": "Kubernetes::Node", "labels": {"aws/account-id": tags['account'], "topology.kubernetes.io/zone": config["ZoneName"]}},
```

File: fixtures/aws.yaml (hunk 6 above)
Comment (adityathebe) [reply fixtures/aws.yaml:46]: The name of kubernetes cluster is user given though.

**Pull request merged**